pkg/envoy/bootstrap/config.go

```diff
@@ -330,43 +331,106 @@ func GetValidationContextSDSConfigYAML() ([]byte, error) {
 // startup, readiness and liveness probes.
 // These will not change during the lifetime of the Pod.
 // If the original probe defined a TCPSocket action, listener and cluster objects are not configured
-// to serve that probe.
+// to serve that probe since the osm-healthcheck container that does that job is configured separately.
 func (b *Builder) getProbeResources() ([]*xds_listener.Listener, []*xds_cluster.Cluster, error) {
 	// This slice is the list of listeners for liveness, readiness, startup IF these have been configured in the Pod Spec
 	var listeners []*xds_listener.Listener
 	var clusters []*xds_cluster.Cluster
 
-	// Is there a liveness probe in the Pod Spec?
-	if b.OriginalHealthProbes.Liveness != nil && !b.OriginalHealthProbes.Liveness.IsTCPSocket {
-		listener, err := getLivenessListener(b.OriginalHealthProbes.Liveness)
-		if err != nil {
-			log.Error().Err(err).Msgf("Error getting liveness listener")
-			return nil, nil, err
-		}
-		listeners = append(listeners, listener)
-		clusters = append(clusters, getLivenessCluster(b.OriginalHealthProbes.Liveness))
+	livenessListenerBuilder := probeListenerBuilder{
+		listenerName: livenessListener,
+		inboundPort:  constants.LivenessProbePort,
+	}
+	readinessListenerBuilder := probeListenerBuilder{
+		listenerName: readinessListener,
+		inboundPort:  constants.ReadinessProbePort,
+	}
+	startupListenerBuilder := probeListenerBuilder{
+		listenerName: startupListener,
+		inboundPort:  constants.StartupProbePort,
 	}
 
-	// Is there a readiness probe in the Pod Spec?
-	if b.OriginalHealthProbes.Readiness != nil && !b.OriginalHealthProbes.Readiness.IsTCPSocket {
-		listener, err := getReadinessListener(b.OriginalHealthProbes.Readiness)
-		if err != nil {
-			log.Error().Err(err).Msgf("Error getting readiness listener")
-			return nil, nil, err
+	for containerName, probes := range b.OriginalHealthProbes {
+		// Is there a liveness probe in the Pod Spec?
+		if probes.Liveness != nil && !probes.Liveness.IsTCPSocket {
```

Comment:
this indicates somewhat deep knowledge of the probeListenerBuilder internals, which makes it fragile to refactoring. Instead, can we simply say `append(livenessListenerBuilder.probes, probes.Liveness)`

And defer the logic of inspecting the isHTTP to the internals of the builder?

Reply:
I think this would also really re-use some of the logic across liveness, readiness, startup

Reply:
The spirit behind this logic already existed, but the idea is that TCPSocket probes don't go through this process at all since they are transformed into an HTTP probe elsewhere

Reply:
I think there's a misunderstanding. I'm simply saying, add a `builder,AddProbe(probes.Liveness)` to the builder, and move the logic present here (which is duplicated across 3 instances), into the actual builder code.

Reply:
to clarify, i'm suggesting the code turns into:


```go
for containerName, probes := range b.OriginalHealthProbes {
  livenessBuilder.AddProbe(probes.Liveness)
  readinessBuilder.AddProbe(probes.Readiness)
  startupBuilder.AddProbe(probes.Startup)
}
```

You can also add the type and the container name as fields to the model.Probe, or simply add extra params to the AddProbe method

Reply:
Ah, I see; agreed 👍🏾